compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/Accessibility.uikit.kt

```diff
@@ -929,13 +991,18 @@ internal class AccessibilityMediator(
     init {
         getAccessibilitySyncOptions().debugLoggerIfEnabled?.log("AccessibilityMediator for ${view} created")
 
-        val updateIntervalMillis = 50L
-        // TODO: this approach was copied from desktop implementation, obviously it has a [updateIntervalMillis] lag
-        //  between the actual change in the semantics tree and the change in the accessibility tree.
-        //  should we use some other approach?
         coroutineScope.launch {
             while (isAlive) {
-                var result: NodesSyncResult
+                var strategy = SemanticsTreeSyncStrategy.from(
```

Comment:
It's not necessary to send this data through the channel.

The channel is just used as a synchronization primitive to trigger the coroutine to awake. The information about what to do when awaking can just be put in a regular data structure by `onSemanticsChange` and `onLayoutChange`.

Reply:
This approach has more straightforward data-flow though instead of scattering the states all around the class.

Reply:
It could affect performance, and IMO is not a good idea in general (mixing a synchronization utility with logic). Tomorrow there could be a better/faster way to trigger a coroutine to wake up (like Java's notify) that doesn't allow data to pass through.

But your call.

Reply:
So you'd insist to revert https://github.com/JetBrains/compose-multiplatform-core/pull/1170/commits/e8de6eee6bb9a3708f7e7042dbbf6d3f49aac418?

Reply:
Reverted

Reply:
I don't insist, just suggesting.